include/ruby/internal/gc.h

```diff
@@ -20,11 +20,372 @@
  *             extension libraries.  They could be written in C++98.
  * @brief      Registering values to the GC.
  */
+#ifdef STDC_HEADERS
+# include <stddef.h>                       /* size_t */
+#endif
+
+#ifdef HAVE_SYS_TYPES_H
+# include <sys/types.h>                    /* ssize_t */
+#endif
+
+#include "ruby/internal/attr/cold.h"
+#include "ruby/internal/attr/nonnull.h"
+#include "ruby/internal/attr/noreturn.h"
+#include "ruby/internal/attr/pure.h"
+#include "ruby/internal/config.h"
```

Comment:
`#include "ruby/internal/config.h"` needs to be the first thing in the file because that's what gives access to `STDC_HEADERS` and `HAVE_SYS_TYPES_H`.